identity/client/src/main/java/com/azure/identity/IdentityClient.java

```diff
@@ -149,19 +149,26 @@ private static AuthenticationContext createAuthenticationContext(ExecutorService
         return context;
     }
 
-    /**
+   /**
      * Asynchronously acquire a token from the App Service Managed Service Identity endpoint.
      * @param msiEndpoint the endpoint to acquire token from
      * @param msiSecret the secret to acquire token with
+     * @param clientId the client ID of the application service
      * @param scopes the scopes to authenticate to
      * @return a Publisher that emits an AccessToken
      */
-    public Mono<AccessToken> authenticateToManagedIdentityEnpoint(String msiEndpoint, String msiSecret, String[] scopes) {
+    public Mono<AccessToken> authenticateToManagedIdentityEnpoint(String msiEndpoint, String msiSecret, String clientId, String[] scopes) {
         String resource = ScopeUtil.scopesToResource(scopes);
         HttpURLConnection connection = null;
         try {
-            String urlString = String.format("%s?resource=%s&api-version=2017-09-01", msiEndpoint, resource);
-            URL url = new URL(urlString);
+            StringBuilder urlStringBuilder = new StringBuilder(String.format("%s?resource=%s&api-version=2017-09-01", msiEndpoint, resource));
```

Comment:
It's weird that you're using string.format within a string builder. You should do only one (stringbuilder I would say).

Reply:
Updated.
StringBuilder not needed here anymore